src/core/Object3D.js

```diff
@@ -472,18 +472,13 @@ THREE.Object3D.prototype = {
 
 	getWorldScale: function () {
 
-		var position = new THREE.Vector3();
-		var quaternion = new THREE.Quaternion();
-
 		return function ( optionalTarget ) {
 
 			var result = optionalTarget || new THREE.Vector3();
 
 			this.updateMatrixWorld( true );
 
-			this.matrixWorld.decompose( position, quaternion, result );
-
-			return result;
+			return result.setFromMatrixScale( this.matrixWorld );
```

Comment:
It's because the decompose generates a -sx at negative determinant. But why isn't that necessary on setFromMatrixScale? @mrdoob